### PR TITLE
Add social-spring-boot post

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

File: _source/_posts/2018-07-23-social-spring-boot.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+---
+layout: blog_post
+title: "Add Social Login Your Spring Boot 2 App"
```
Comment (mraible) [inline _source/_posts/2018-07-23-social-spring-boot.md]: Is 2.0 better than just 2? Not sure.

File: _source/_posts/2018-07-23-social-spring-boot.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+author: moksamedia
+description: "The primary benefits of social login are ease of use and security. It's easier for your users to log into your app if they can use credentials they've already created on a a service they trust."
+tags: [authentication, spring boot, oidc, social]
```
Comment (mraible) [inline _source/_posts/2018-07-23-social-spring-boot.md]: I’d change “social” to “social login”

File: _source/_posts/2018-07-23-social-spring-boot.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+You'll need the following:
+ - Okta application client ID
+ - Okta application client secret
+ - Your Okta preview URL. Something like: `dev-123456.oktapreview.com`
+ - Your Okta issuer URL. From the Okta dashboard, select **API** in the top menu and then click **Authorization Servers**, and you'll see the **Issuer URI** for the default authorization server
+
+You'll want to check your Okta OIDC Application's configuration. It should match the following photo. This is also where you'll get your client ID and client secret. **Note that under "Allowed grant types" the "Implicit (Hybrid)" checkbox is checked (it's not checked by default).**
+
+{% img blog/social-spring-boot/enable-implcit.png alt:"Enable Implicit Flow" width:"600" %}{: .center-image }
```
Comment (mraible) [inline _source/_posts/2018-07-23-social-spring-boot.md]: Probably doesn't matter, but there's a typo in implicit

File: _source/_posts/2018-07-23-social-spring-boot.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+You should be able to run `./mvnw` from the terminal to run the app (you only need to type `./mvnw` because the `pom.xml` specifies a default goal of `spring-boot:run`, so typing `./mvnw` is the same as typing `./mvnw spring-boot:run`). 
+
+To test the app, **log out of your developer.okta.com account** and go to `http://localhost:8080` (Or open an incognito window).
```
Comment (mraible) [inline _source/_posts/2018-07-23-social-spring-boot.md]: Or > or

File: _source/_posts/2018-07-23-social-spring-boot.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+Your Okta Identity Providers page should now look like the following. Note the two "IdP IDs" - you'll need these in a moment.
+
+{% img blog/social-spring-boot/social-idps-added.png alt:"Add Facebook IdP" width:"800" %}{: .center-image }
+
+## Customize the Self-Hosted Sign-In Page for Your Spring Boot App
+
+The final step is to update the Okta Sign-in Widget on your customized sign-in page. This part is ridiculously easy.  You'll need the IdP IDs from above. 
+
+Open the `src/main/resources/templates/login.html` file and take a look at the javascript block at the bottom of the page. You configure the identity providers in the Okta Sign In widget's config using the `idps` property.
```
Comment (mraible) [inline _source/_posts/2018-07-23-social-spring-boot.md]: javascript > JavaScript

Comment (mraible) [reply _source/_posts/2018-07-23-social-spring-boot.md]: Okta Sign In widget's > Okta Sign-In Widget's

File: _source/_posts/2018-07-23-social-spring-boot.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+## Customize the Self-Hosted Sign-In Page for Your Spring Boot App
+
+The final step is to update the Okta Sign-in Widget on your customized sign-in page. This part is ridiculously easy.  You'll need the IdP IDs from above. 
+
+Open the `src/main/resources/templates/login.html` file and take a look at the javascript block at the bottom of the page. You configure the identity providers in the Okta Sign In widget's config using the `idps` property. 
+
+See the commented section of code below.
+
+```js
+config.baseUrl = /*[[${oktaBaseUrl}]]*/ '{oktaBaseUrl}';
```
Comment (mraible) [inline _source/_posts/2018-07-23-social-spring-boot.md]: /*[[${oktaBaseUrl}]]*/ '{oktaBaseUrl}';  - this a strange syntax I haven't seen before. Shouldn't this be `{yourOktaDomain}`?

File: _source/_posts/2018-07-23-social-spring-boot.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+  },  
+  function (err) {  
+    console.log(err)  
+  }  
+);
+```
+
+That's it! Pretty cool, huh?
+
+The Okta Sign-In Widget has a lot of great configuration options, and you can check out the docs for them [on the Github page](https://github.com/okta/okta-signin-widget).
```
Comment (mraible) [inline _source/_posts/2018-07-23-social-spring-boot.md]: Github > GitHub

File: _source/_posts/2018-07-23-social-spring-boot.md (hunk 6 above)
Comment (bdemers) [reply _source/_posts/2018-07-23-social-spring-boot.md]: I'm not sure on this one, this was some Thymeleaf work around IIRC. the `'{oktaBaseUrl}'` part is never rendered.  I don't know what the _best_ option here is.

I know @dogeared didn't go this route on his last project either.

What do you guy do when filtering into JS from Thymeleaf?

**Pull request merged**